### PR TITLE
Add destructor to Fmod wrapper.

### LINKED ISSUE #58: Memory leakage when not initializing Fmod
I'm getting some memory leakage in my project, but only when I don't call `Fmod.init()`

image

This happens because the Fmod GDNative instance is created on _ready(), but not freed automatically when Fmod.gd exists the tree.

**A possible solution:**
Free the `fmod_godot` variable (containing the GDNative instance) in an `_exit_tree()`-function as such:
image

This seems to get rid of the memory leakage.

NOTE: I haven't thoroughly tested this! There might be issues when you are actually using Fmod! (This could be solved by adding an if-loop that checks if the `fmod_godot` variable is null or not)

**How to replicate:**
Just add the FMOD addons folder and set `verbose_output` to true in the ProjectSettings. Running the game and then quitting it (by pressing the X button) will nag about memory leakage.

### PR DESCRIPTION
**Pull request by CedNaru**
Fix #58 

The autoload wrapper now has a destructor that will free the GDNative instance. 
No leak should happen when the game is closing with this fix.

### CODE REVIEW

**Comment by CedNaru** [comment]
Now Fmod is shutdown, if started, before the GDnative instance is freed

File: src/godot_fmod.cpp [hunk 1]
```diff
@@ -11,7 +11,6 @@ Mutex *Callbacks::mut;
 Fmod::Fmod() = default;
 
 Fmod::~Fmod() {
-    Fmod::shutdown();
```
Comment (piiertho) [inline src/godot_fmod.cpp:14]: What if someone wants to use the library without the GDScript wrapper ?
Maybe we can have a check here to handle if already shutdown or not ?

Comment (CedNaru) [reply src/godot_fmod.cpp:14]: Well they have to shutdown FMOD themselves before imo.
Fmod is not initialized in the constructor of the class so I don't think it should be the job of the destructor to shutdown it.
If we still want to have a check, we have to move the logic from GDScript to C++.

Comment (piiertho) [reply src/godot_fmod.cpp:14]: I'm fine with it. Maybe we can improve it later

**Review by piiertho — APPROVED** [review]

**Pull request merged**